server/errors.go

```diff
@@ -22,11 +22,14 @@ var (
 	// ErrConnectionClosed represents an error condition on a closed connection.
 	ErrConnectionClosed = errors.New("Connection Closed")
 
-	// ErrAuthorization represents an error condition on failed authorization.
-	ErrAuthorization = errors.New("Authorization Error")
+	// ErrAuthorization represents an error condition on failed authentication.
+	ErrAuthentication = errors.New("Authentication Error")
```

Comment:
This is going to break client libraries. For instance go-nats checks error to process auth error in certain way: https://github.com/nats-io/go-nats/blob/master/nats.go#L2197

Reply:
Agree we do not send this to client. Tested that today.

Reply:
Yes, I verified that we actually hardcode the value that we send to client in `authViolation()` by calling:
```
c.sendErr("Authorization Violation")
```